test/test_package_loading.py

```diff
@@ -34,3 +35,30 @@ def test_invalid_package_name(selenium):
     with pytest.raises(WebDriverException,
                        match="Invalid package name or URI"):
         selenium.load_package('tcp://some_url')
+
+
+def test_load_packages_multiple(selenium_standalone):
+    selenium = selenium_standalone
+    selenium.load_package(['pyparsing', 'matplotlib'])
+    selenium.run('import pyparsing')
+    selenium.run('import matplotlib')
+    assert selenium.logs.count('Loading pyparsing') == 1
+
+
+@pytest.mark.xfail(reason='Not implemented')
+def test_load_packages_simultaneous(selenium_standalone):
+    selenium = selenium_standalone
+
+    from selenium.common.exceptions import TimeoutException
+
+    selenium.run_js(
+        'window.done = false\n'
+        'pyodide.loadPackage("numpy")\n'
+        'pyodide.loadPackage("matplotlib")'
+        '.then(function() { window.done = true; })')
```

Comment:
This sort of thing *could* get flaky on Chrome. On Chrome, we are forced to load the wasm asynchronously, but unfortunately that uses global variables to set up, so you can't actually load wasm modules simultaneously.  All that happens in emscripten itself, but I don't think it actually does any locking around simultaneous loading.

We may want to add our own locking around this case.  i.e., if there's already an active promise, attach the new promise to run *after* that promise.

Reply:
Thanks for the review and your comments !  I responded in https://github.com/iodide-project/pyodide/issues/135#issuecomment-418402486 and removed this failing test: as I haven't found a solution for this yet, I would rather make a subsequent PR for it...